flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/FlinkDistribution.java

```diff
@@ -301,7 +309,105 @@ public void submitSQLJob(SQLJobSubmission job, Duration timeout) throws Exceptio
                             });
             new Thread(future).start();
             future.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
+        } else if (job.getClientMode() == SQLJobSubmission.ClientMode.REST) {
+            FutureTaskWithException<Void> future =
+                    new FutureTaskWithException<>(
+                            () -> {
+                                Map<String, String> configMap =
+                                        GlobalConfiguration.loadConfiguration(
+                                                        conf.toAbsolutePath().toString())
+                                                .toMap();
+                                String host =
+                                        configMap.getOrDefault(
+                                                "sql-gateway.endpoint.rest.address",
+                                                InetAddress.getByName("localhost")
+                                                        .getHostAddress());
+                                String port =
+                                        configMap.getOrDefault(
+                                                "sql-gateway.endpoint.rest.port", "8083");
+                                // Open a session
+                                String sessionHandle = openSession(host, port);
+                                List<String> sqlLines = job.getSqlLines();
+                                for (String sql : sqlLines) {
+                                    // Execute statement
+                                    String operationHandle =
+                                            executeStatement(sessionHandle, sql, host, port);
+                                    int count = 5;
+                                    while (count > 0) {
+                                        TimeUnit.SECONDS.sleep(2);
```

Comment:
It may not be a good idea to directly sleep for 2 seconds at the beginning. I am worried that this will affect the running time of E2E CI. If we can't think of a good way to deal with this problem, at least we can move the sleep statement after getting the results and reduce the sleep time appropriately.

Reply:
Has been modified to keep fetching data until no data is generated.